File: Cano-Rodriguez-Claudio/Procesos/Java/process.java
public class ProcessCountdown {
    static final int MAX = 5; // Countdown limit

    public static void main(String[] args) {
        for (int i = 1; i <= 3; i++) {
            Thread child = new Thread(() -> {
                for (int k = MAX; k >= 1; k--) {
                    System.out.println("Child Thread [" + Thread.currentThread().getId() + "]: Countdown = " + k);
                    try {
                        Thread.sleep(500); // Sleep for 500ms
                    } catch (InterruptedException e) {
                        System.err.println("Error: Sleep interrupted");
                    }
                }
            });
            child.start();
        }
    }
}

File: Cano-Rodriguez-Claudio/Procesos/Go/process.go
package main

import (
	"fmt"
	"os"
	"sync"
	"time"
)

const MAX = 5 // Countdown limit

func countdown(wg *sync.WaitGroup, id int) {
	defer wg.Done()
	for k := MAX; k >= 1; k-- {
		fmt.Printf("Child Process [%d]: Countdown = %d\n", os.Getpid(), k)
		time.Sleep(500 * time.Millisecond)
	}
}

func main() {
	var wg sync.WaitGroup

	for i := 1; i <= 3; i++ {
		wg.Add(1)
		go countdown(&wg, i)
	}

	wg.Wait()
}
